### ISSUE TITLE
Special characters@ add search box

### ISSUE THREAD

**Issue by rgreen5** [musescore/MuseScore#23666]
### Your idea

Add a search function to the Special characters window.

### Problem to be solved

This would make finding characters in the "Musical symbols" and "Unicode symbols" possible. At the moment it is almost impossible to  find anything unless it is in the "Common" window, and this limits the usefulness of the SC palette.

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by bkunda** [comment]
@avvvvve did you already have a design for this?

**Comment by rgreen5** [comment]
No doubt this has already been taken into account, but you also need to be able to _delete_ characters from the "Common" palette as well.

**Comment by avvvvve** [comment]
@bkunda yes, but as part of an overhauled design of the special characters window to turn it into a popup, which is logged here: https://github.com/musescore/MuseScore/issues/21059

if someone wants to add a search to the existing dialog as a temporary solution, by all means they should!

**Comment by RMRattray** [comment]
Hello,
I am interested in making general contributions; would this be a good thing for me to be assigned to start working on?

**Comment by cbjeukendrup** [comment]
Given [the plan mentioned above](https://github.com/musescore/MuseScore/issues/21059) to rewrite the entire 'special characters' dialog as a popup, this might not be the most convenient issue to work on right now. But you could reach out to @avvvvve at that issue, to discuss the possibility of contributing the popup implementation. That might be a bit challenging for a somewhat-first contribution, though.
